code/entities/machine/soda-fountain/interactable/Platform.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System.Linq;
+using Sandbox;
+using Sandbox.util;
+using Conna.Inventory;
+
+namespace Cinema;
+
+public class Platform : BaseInteractable
+{
+    static private int NumSlots = 3;
+
+    private Slot[] Slots = new Slot[NumSlots];
+    public Platform() 
+    {
+        for (int i = 0; i < NumSlots; i++)
+        {
+            Slots[i] = new Slot(i, $"lever{i + 1}_cup");
+        }
+    }
+
+    /// <summary>
+    /// This will add or take the cup you are pressing closest to, if nothing is close to it, it will add a cup to the nearest possible slot.
+    /// </summary>
+    /// <param name="player"></param>
+    public override void Trigger(Player player)
+    {
+        var slot = GetClosestSlot(player);
+
+        if (slot.HasItem())
+        {
+            TakeCup(slot, player);
+        }
+
+        if (slot.IsEmpty())
```

Comment:
Wouldn't this take the cup and immediately put a new (empty) one?

Should we return right after line 31 to prevent this?

Reply:
Added a return for this